File: 02_selection_sort/selection_sort.rb
def selecttion_sort(list)
  current_i = 0

  while current_i < list.size - 1
    min_i = current_i

    list[current_i..-1].each_with_index do |n, i|
      min_i = list[min_i] < n ? min_i : (i + current_i)
    end

    list[current_i], list[min_i] = list[min_i], list[current_i]

    current_i += 1
  end

  list
end

pp selecttion_sort [*1..10].shuffle

File: 02_selection_sort/selection_sort.ts
const selection_sort = <T>(list: T[]): T[] => {
  let current_i: number = 0

  while(current_i < list.length - 1) {
    let min_i: number = current_i
    
    list.slice(current_i).forEach((n, i) => {
      min_i = list[min_i] < n ? min_i : (i + current_i)
    })

    let [curr, min] = [list[current_i], list[min_i]];
    list[current_i] = min;
    list[min_i] = curr;

    current_i++
  }

  return list
}


const list = Array.from({length: 10}, () => Math.floor(Math.random() * 40));
const sorted = selection_sort(list)

console.log(sorted)
